File: subprojects/frida-python/examples/inject_library/inject_file.py
#
# Compile example.dylib like this:
# $ clang -shared example.c -o ~/.Trash/example.dylib
#
# Then run:
# $ python inject_file.py Twitter ~/.Trash/example.dylib
#

import sys

import frida


def on_uninjected(id):
    print("on_uninjected id=%u" % id)


target, library_path = sys.argv[1:]

device = frida.get_local_device()
device.on("uninjected", on_uninjected)
id = device.inject_library_file(target, library_path, "example_main", "w00t")
print("*** Injected, id=%u -- hit Ctrl+D to exit!" % id)
sys.stdin.read()

File: subprojects/frida-node/examples/inject_library/inject_file.ts
/*
 * Compile example.dylib like this:
 * $ clang -shared example.c -o ~/.Trash/example.dylib
 *
 * Then run:
 * $ node inject_file.js Twitter ~/.Trash/example.dylib
 */

import frida from "frida";

const [target, libraryPath] = process.argv.slice(2);

let device: frida.Device | null = null;

async function main() {
    device = await frida.getLocalDevice();
    device.uninjected.connect(onUninjected);

    try {
        const id = await device.injectLibraryFile(target, libraryPath, "example_main", "w00t");
        console.log("[*] Injected id:", id);
    } catch (e) {
        device.uninjected.disconnect(onUninjected);
        throw e;
    }
}

function onUninjected(id: frida.InjecteeID) {
    console.log("[*] onUninjected() id:", id);
    device!.uninjected.disconnect(onUninjected);
}

main()
    .catch(e => {
        console.error(e);
    });
